pandas/tests/plotting/test_misc.py

```diff
@@ -441,6 +442,29 @@ def test_dictionary_color(self):
         colors = [rect.get_color() for rect in ax.get_lines()[0:2]]
         assert all(color == expected[index] for index, color in enumerate(colors))
 
+    def test_bar_plot(self):
+        # GH38947
+        # Test bar plot with string and int index
+        from matplotlib.text import Text
+
+        expected = [Text(0, 0, "0"), Text(1, 0, "Total")]
+
+        df = DataFrame(
+            {
+                "a": [1, 2],
+            },
+            index=Index([0, "Total"]),
+        )
+
+        try:
```

Comment:
You don't need the try/except. The `assert` will raise appropriately.

Reply:
Done, thanks! Getting a timeout error after 90min in the CI but that's another problem